lib/spack/docs/index.rst

```diff
@@ -63,9 +63,9 @@ or refer to the full manual below.
    mirrors
    module_file_support
    repositories
+   binary_caches
    command_index
    package_list
-   binary_caches
```

Comment:
The last 2 sections here are lists of packages and links to other docs, not the kind of documentation that people read. I moved Build Caches above these, as people are likely to skip Command Index and Package List when reading the docs.